src/geometry/cached_system.hpp

```diff
@@ -456,93 +479,98 @@ void SetCachedCoordinateSystem(MeshBlockData<Real> *rc) {
+  IndexRange ib = rc->GetBoundsI(IndexDomain::entire);
+  IndexRange jb = rc->GetBoundsJ(IndexDomain::entire);
+  IndexRange kb = rc->GetBoundsK(IndexDomain::entire);
   int kbs = axisymmetric ? 0 : kb.s;
   int kbe = axisymmetric ? 0 : kb.e;
-  pmb->par_for(
-      "SetGeometry::Set Cached data, Cent", kbs, kbe, jb.s, jb.e, ib.s, ib.e,
-      KOKKOS_LAMBDA(const int k, const int j, const int i) {
-        lamb(k, j, i, CellLocation::Cent);
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "SetGeometry::Set Cached data, Cent", DevExecSpace(), 0,
+      pack.GetDim(5) - 1, kbs, kbe, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        lamb(b, k, j, i, CellLocation::Cent);
       });
-  pmb->par_for(
-      "SetGeometry::Set Cached data, Face1", kbs, kbe, jb.s, jb.e, ib.s, ib.e + 1,
-      KOKKOS_LAMBDA(const int k, const int j, const int i) {
-        lamb(k, j, i, CellLocation::Face1);
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "SetGeometry::Set Cached data, Face1", DevExecSpace(), 0,
+      pack.GetDim(5) - 1, kbs, kbe, jb.s, jb.e, ib.s, ib.e + 1,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        lamb(b, k, j, i, CellLocation::Face1);
       });
-  pmb->par_for(
-      "SetGeometry::Set Cached data, Face2", kbs, kbe, jb.s, jb.e + 1, ib.s, ib.e,
-      KOKKOS_LAMBDA(const int k, const int j, const int i) {
-        lamb(k, j, i, CellLocation::Face2);
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "SetGeometry::Set Cached data, Face2", DevExecSpace(), 0,
+      pack.GetDim(5) - 1, kbs, kbe, jb.s, jb.e + 1, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        lamb(b, k, j, i, CellLocation::Face2);
       });
   if (!axisymmetric) kbe = kb.e + 1;
-  pmb->par_for(
-      "SetGeometry::Set Cached data, Face3", kbs, kbe, jb.s, jb.e, ib.s, ib.e,
-      KOKKOS_LAMBDA(const int k, const int j, const int i) {
-        lamb(k, j, i, CellLocation::Face3);
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "SetGeometry::Set Cached data, Face3", DevExecSpace(), 0,
+      pack.GetDim(5) - 1, kbs, kbe, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        lamb(b, k, j, i, CellLocation::Face3);
       });
-  pmb->par_for(
-      "SetGeometry::Set Cached data, Corn", kbs, kbe, jb.s, jb.e + 1, ib.s, ib.e + 1,
-      KOKKOS_LAMBDA(const int k, const int j, const int i) {
-        lamb(k, j, i, CellLocation::Corn);
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "SetGeometry::Set Cached data, Corn", DevExecSpace(), 0,
+      pack.GetDim(5) - 1, kbs, kbe, jb.s, jb.e + 1, ib.s, ib.e + 1,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        lamb(b, k, j, i, CellLocation::Corn);
       });
 
   // don't let other kernels launch until geometry is set
-  pmb->exec_space.fence();
+  Kokkos::fence();
```

Comment:
do we actually need this fence?

Reply:
I could be convinced to drop it... My concern was a race condition where the geometry variables are accessed before they're filled. i.e., if you call `pgen` before the geometry vars are done.